client.go

```diff
@@ -110,7 +110,7 @@ func NewClient[Req, Res any](httpClient HTTPClient, url string, options ...Clien
 		request.peer = client.protocolClient.Peer()
 		protocolClient.WriteRequestHeader(StreamTypeUnary, request.Header())
 		response, err := unaryFunc(ctx, request)
-		if err != nil {
+		if err != nil || response == nil {
```

Comment:
This doesn't seem right. A `nil, nil` response isn't really valid. What is the framework expected to do with this when returning a value to the caller?

Reply:
See https://github.com/connectrpc/connect-go/pull/799#discussion_r1868507259